pkg/controller/direct/sql/mapping.go

```diff
@@ -23,13 +23,17 @@ import (
 	krm "github.com/GoogleCloudPlatform/k8s-config-connector/apis/sql/v1beta1"
 )
 
-func SQLInstanceKRMToGCP(in *krm.SQLInstance, refs *SQLInstanceInternalRefs) (*api.DatabaseInstance, error) {
+func SQLInstanceKRMToGCPInstance(in *krm.SQLInstance, refs *SQLInstanceInternalRefs) (*api.DatabaseInstance, error) {
 	out := &api.DatabaseInstance{}
 
 	if in == nil {
 		return nil, fmt.Errorf("cannot convert nil SQLInstance")
 	}
 
+	if in.Spec.CloneSource != nil {
+		return nil, fmt.Errorf("cannot convert SQLInstance with CloneSource specified")
```

Comment:
~Does it mean that while the field is supported in the CRD, the feature can't be used?~

Okay, so basically this conversion is only triggered by the insert function, and the insert function should never be called if `in.Spec.CloneSource != nil`... got it.

Maybe the error message can be improved in the caller (insertInstance() function) saying something like: `error inserting a Cloud SQL instance with no cloud source: %w`, or you can make the error message more verbose here; otherwise this is an implicit knowledge which is not intuitive to future contributors.

Reply:
This error would only happen if developer misuses the conversion function. Probably just adding a comment would be sufficient.

Reply:
fixed